src/ansiblelint/rules/name.py

```diff
@@ -0,0 +1,56 @@
+"""Implementation of NameRule."""
+import sys
+from typing import TYPE_CHECKING, Any, Dict, Union
+
+from ansiblelint.errors import MatchError
+from ansiblelint.rules import AnsibleLintRule
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    from ansiblelint.file_utils import Lintable
+
+
+class NameRule(AnsibleLintRule):
+    """All tasks should be named."""
+
+    id = "name"
+    description = (
+        "All tasks should have a distinct name for readability "
+        "and for ``--start-at-task`` to work"
+    )
+    severity = "MEDIUM"
+    tags = ["idiom"]
+    version_added = "historic"
+
+    def matchtask(
+        self, task: Dict[str, Any], file: "Optional[Lintable]" = None
+    ) -> Union[bool, str, MatchError]:
+        if not task.get("name"):
+            return self.create_matcherror(
+                linenumber=task["__line__"], tag="name[missing]", filename=file
```

Comment:
Let's use the LINE_NUMBER_KEY constant in ansiblelint.utils.LINE_NUMBER_KEY

Reply:
Also, this bypasses some of the enrichment that happens in `matchtasks()` like `details` and `match.task`.
https://github.com/ansible/ansible-lint/blob/a5f0d521e404ae8839d5879b20fb44200aeca1f9/src/ansiblelint/rules/__init__.py#L162-L177

I'm not sure where we should add that. Maybe `matchtasks()` should be refactored to update `details` if it is not set, and add `match.task`.

Reply:
I was aware about missing details but I was not sure what to include in them. The reality is that the generic "name" description is no longer relevant for various violations.

Probably we need to add some longer descriptions but we need to write the text for them.